### ISSUE TITLE
Can't schedule an import when the webhook is a WordPress-based site

### ISSUE THREAD

**Issue by MegaphoneJon** [4ndygu/civicrm_osdi#18]
The various CMSes that CiviCRM supports all have different methods of indicating to their front page router which page should be loaded.  Here are two examples:

```
Drupal: https://test-drupal.megaphonetech.com/civicrm/osdi/webhook
WordPress: https://testsite-wp.megaphonetech.com/wp-admin/admin.php?page=CiviCRM&q=civicrm%2Fosdi%2Fwebhook
```

Furthermore, this assumes that Drupal "clean URLs" are enabled.  Otherwise, the Drupal URLs are constructed similarly:

```
Drupal (no clean URLs): http://example.org/index.php?q=civicrm/osdi/webhook
Joomla 1.5 sites: http://example.org/administrator/index2.php?option=com_civicrm&task=civicrm/osdi/webhook
Joomla 1.6+ sites: http://example.org/administrator/index.php?option=com_civicrm&task=civicrm/osdi/webhook
WordPress sites: http://example.org/wp-admin/admin.php?page=CiviCRM&q=civicrm/osdi/webhook
```

If it helps to troubleshoot, I provisioned a WordPress site at https://osdi3.megaphonetech.com on the same server as the other two.

**Comment by 4ndygu** [comment]
Hmm, to fix this - is there an environment variable I can use to pull the CMS I'm using?

**Comment by MegaphoneJon** [comment]
I think the problem is the URL of the webhook on the remote.  I set up https://osdi3.megaphonetech.com on the same server (running WordPress/Civi) to check

**Comment by 4ndygu** [comment]
I agree too - my functionality right now is just to take the civicrm root and append the Drupal URL. Is there some way for me to grab the CMS version and provider so I can provide custom generation of the URL?

**Comment by MegaphoneJon** [comment]
Ah - I think my confusion is the "OSDI Endpoint (root)" is actually looking for the CiviCRM root, and not the actual OSDI Endpoint.  I think we could bypass this issue altogether if we just required the OSDI Endpoint itself.  I assume that wouldn't cause an issue with Action Network integration?

For usability purposes, we should also add some code that displays the OSDI endpoint for the local server.  You can get that with a call to `CRM_Utils_System::url('civicrm/osdi/config')`.  There's an equivalent Smarty call you can make, which you can see by grepping the core codebase for `crmURL` in the `templates` directory.

**Comment by j-ro** [comment]
Agree here, just having folks type in a URL I think is fine, as long as it's documented with a Civi URL will be.

**Comment by 4ndygu** [comment]
i'll write a PR to reformat the URLs - quick q: for the wordpress site, how would I pass in further parameters to the webhook URL? a URL encoded '?'?

**Comment by 4ndygu** [comment]
also, in the api/v3/Export/Export.php file, it looks like I generate URL's on the fly for the next "page" of users, per the OSDI spec. Is there a way to generate these URLs on the fly while accounting for the different CMSes?
